From 28968bd3122535345c8fad2e9b7e7dd71059cbd4 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 17 Oct 2023 19:41:58 +0100
Subject: [PATCH 1/1] clang/lib: Ignore Werror if CLANG_IGNORE_WERROR is set.

Useful when projects force append -Werror to FLAGS

Downstream, Solus only.
---
 clang/lib/Basic/Warnings.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/clang/lib/Basic/Warnings.cpp b/clang/lib/Basic/Warnings.cpp
index cc8c13823..6dc3212f2 100644
--- a/clang/lib/Basic/Warnings.cpp
+++ b/clang/lib/Basic/Warnings.cpp
@@ -25,6 +25,7 @@
 #include "clang/Basic/Diagnostic.h"
 #include "clang/Basic/DiagnosticOptions.h"
 #include <algorithm>
+#include <cstdlib>
 #include <cstring>
 #include <utility>
 using namespace clang;
@@ -148,8 +149,9 @@ void clang::ProcessWarningOptions(DiagnosticsEngine &Diags,
 
         if (Specifier.empty()) {
           if (SetDiagnostic)
-            Diags.setWarningsAsErrors(isPositive);
-          continue;
+            if (!getenv("CLANG_IGNORE_WERROR"))
+              Diags.setWarningsAsErrors(isPositive);
+            continue;
         }
 
         if (SetDiagnostic) {
-- 
2.42.0
